src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1537,17 +1542,24 @@ public async Task Dispose_ProcessingResponse_OK()
 
                 // Start streaming response and wait for client to be disposed.
                 byte[] responseBody = new byte[100];
-                while (!diposeCalled)
+                while (!disposeCalled)
```

Comment:
I don't really understand why we do this. Do you know? Seems like a weird thing for the test to do.

Reply:
It seems like what this test should do is:

(1) issue a request and get back the response message, but don't read it yet
(2) dispose the HttpClient
(3) signal the server to send some data -- a known amount of data < 64K, so we don't need to worry about control flow
(4) retrieve the data on the client and validate it

Reply:
perhaps non-deterministic factor checks more scenarios 😄 Will update per recommendation